subsys/bluetooth/controller/ll_sw/ull_sync_iso.c

```diff
@@ -42,93 +42,138 @@
+	if (!link_sync_estab) {
+		return BT_HCI_ERR_MEM_CAPACITY_EXCEEDED;
+	}
+
+	link_sync_lost = ll_rx_link_alloc();
+	if (!link_sync_lost) {
+		ll_rx_link_release(link_sync_estab);
+
 		return BT_HCI_ERR_MEM_CAPACITY_EXCEEDED;
 	}
 
-	big_sync_estab = ll_rx_link_alloc();
-	if (!big_sync_estab) {
+	node_rx = ll_rx_alloc();
+	if (!node_rx) {
+		ll_rx_link_release(link_sync_lost);
+		ll_rx_link_release(link_sync_estab);
+
 		return BT_HCI_ERR_MEM_CAPACITY_EXCEEDED;
 	}
 
-	big_sync_lost = ll_rx_link_alloc();
-	if (!big_sync_lost) {
-		ll_rx_link_release(big_sync_estab);
+	sync_iso = sync_iso_acquire();
+	if (!sync_iso) {
+		ll_rx_release(node_rx);
+		ll_rx_link_release(link_sync_lost);
+		ll_rx_link_release(link_sync_estab);
 
 		return BT_HCI_ERR_MEM_CAPACITY_EXCEEDED;
 	}
 
-	sync_iso->node_rx_lost.link = big_sync_lost;
-	sync_iso->node_rx_estab.link = big_sync_estab;
+	/* Initialize the ISO sync ULL context */
+	sync_iso->sync = sync;
+	sync_iso->node_rx_lost.hdr.link = link_sync_lost;
 
-	lll_sync = &sync_iso->lll;
+	/* Setup the periodic sync to establish ISO sync */
+	node_rx->link = link_sync_estab;
+	sync->iso.node_rx_estab = node_rx;
 
-	/* Initialise ULL and LLL headers */
+	/* Initialize ULL and LLL headers */
 	ull_hdr_init(&sync_iso->ull);
-	lll_hdr_init(lll_sync, sync);
-	lll_sync->is_enabled = true;
+	lll_hdr_init(&sync_iso->lll, sync);
+
+	/* Enable periodic advertising to establish ISO sync */
+	sync->iso.sync_iso = sync_iso;
 
 	return BT_HCI_ERR_SUCCESS;
 }
 
-uint8_t ll_big_sync_terminate(uint8_t big_handle)
+uint8_t ll_big_sync_terminate(uint8_t big_handle, void **rx)
 {
-	memq_link_t *big_sync_estab;
-	memq_link_t *big_sync_lost;
 	struct ll_sync_iso *sync_iso;
+	struct node_rx_pdu *node_rx;
+	memq_link_t *link_sync_estab;
+	memq_link_t *link_sync_lost;
+	struct ll_sync_set *sync;
 	int err;
 
 	sync_iso = ull_sync_iso_get(big_handle);
 	if (!sync_iso) {
 		return BT_HCI_ERR_UNKNOWN_ADV_IDENTIFIER;
 	}
 
-	if (!sync_iso->lll.is_enabled) {
-		return BT_HCI_ERR_CMD_DISALLOWED;
+	sync = sync_iso->sync;
+	if (sync) {
+		struct node_rx_sync_iso *se;
+
+		if (sync->iso.sync_iso != sync_iso) {
+			return BT_HCI_ERR_CMD_DISALLOWED;
+		}
+
+		node_rx = (void *)sync->iso.node_rx_estab;
+		link_sync_estab = node_rx->hdr.link;
+		link_sync_lost = sync_iso->node_rx_lost.hdr.link;
+
+		ll_rx_link_release(link_sync_lost);
+		ll_rx_link_release(link_sync_estab);
+		ll_rx_release(node_rx);
+
+		node_rx = (void *)&sync_iso->node_rx_lost;
+		node_rx->hdr.type = NODE_RX_TYPE_SYNC_ISO;
+		node_rx->hdr.handle = 0xffff;
+
+		/* NOTE: struct node_rx_lost has uint8_t member following the
+		 *       struct node_rx_hdr to store the reason.
+		 */
+		se = (void *)node_rx->pdu;
+		se->status = BT_HCI_ERR_OP_CANCELLED_BY_HOST;
+
+		/* NOTE: Since NODE_RX_TYPE_SYNC is only generated from ULL
+		 *       context, pass ULL context as parameter.
+		 */
+		node_rx->hdr.rx_ftr.param = sync_iso;
+
+		*rx = node_rx;
+
+		return BT_HCI_ERR_SUCCESS;
```

Comment:
If we return here, when/how is the ticker stopped? And do we not still need to release the `link_sync_lost` and `sync_iso` in any case?